src/components/LoaderSpinner.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div class="loader-outer" v-show="isLoading">
+    <ProgressSpinner />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed } from 'vue';
+import ProgressSpinner from 'primevue/progressspinner';
+import { HTTP, HTTP_AUTH } from '@/core/api/http-common';
+import { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios';
+
+export default defineComponent({
+  name: 'LoaderSpinner',
+  components: {
+    ProgressSpinner,
+  },
+  setup() {
+    const isLoading = ref(false);
+
+    computed(() => {
+      return isLoading.value;
+    });
+
+    const showLoader = () => {
+      isLoading.value = true;
+    };
+    const hideLoader = () => {
+      isLoading.value = false;
+    };
+
+    HTTP.interceptors.request.use(
+      (req: AxiosRequestConfig) => {
+        showLoader();
+        return req;
+      },
+      (error: AxiosError) => {
+        hideLoader();
+        return error;
+      }
+    );
+
+    HTTP.interceptors.response.use(
+      (response: AxiosResponse) => {
+        hideLoader();
+        return response;
+      },
+      (error: AxiosError) => {
+        hideLoader();
+        return error;
+      }
+    );
+
+    HTTP_AUTH.interceptors.request.use(
+      (req: AxiosRequestConfig) => {
+        showLoader();
+        return req;
+      },
+      (error: AxiosError) => {
+        hideLoader();
+        return error;
+      }
+    );
+
+    HTTP_AUTH.interceptors.response.use(
```

Comment:
Do you need all this interceptors to be initialized inside the component? How many interceptors will be initialized when you'll create two or three spinner components?